drv/stm32xx-sys-api/src/lib.rs

```diff
@@ -32,6 +33,18 @@ pub enum RccError {
     NoSuchPeripheral = 1,
 }
 
+/// Configures edge sensitivity for a GPIO interrupt
+#[derive(Copy, Clone, FromPrimitive, PartialEq, Eq, AsBytes)]
+#[repr(u8)]
```

Comment:
Nit: not necessary, the use of `FromPrimitive` makes the interface repr-independent.

(If there's some other reason you want this to be `repr(u8)` specifically then carry on.

Reply:
the `repr(u8)` is used for casting to `u8` in these methods: https://github.com/oxidecomputer/hubris/pull/1706/files#diff-7d4882bb0fb1799e94fee962d08ea3adcf32a27babfd1c42bcefa60846faf897R302-R312

we could implement those using `matches!` instead, or something, but i thought this was cuter --- my naive assumption was that it would also generate less code, but this may not be true on thumb targets...

Reply:
update on that: i stuck it into Godbolt and it looks like the bitwise version does, in fact, generate much less code on `thumbv7m` and `thumbv6m`: https://godbolt.org/z/j5fdPfz3c

Reply:
Cool -- maybe add a comment to that effect on the repr attribute, because otherwise it might get removed.